accord-core/src/main/java/accord/coordinate/CheckOn.java

```diff
@@ -94,23 +94,39 @@ protected AbstractRoute route()
     @Override
     protected boolean isSufficient(Id from, CheckStatusOk ok)
     {
-        return ((CheckStatusOkFull)ok).sufficientFor(route()).compareTo(sufficient) >= 0;
+        KeyRanges rangesForNode = topologies().computeRangesForNode(from);
+        PartialRoute scope = this.route.slice(rangesForNode);
```

Comment:
If we don't use the route that was provided to us to check sufficiency then doesn't the check that we have *alll* the information for the route no longer work?

I don't get how it works in general just because the `CheckStatusOK` we are checking is a fraction of the information we have collected and `CheckOn` maybe be trying to cover more shards so we might have sufficient information, but never notice with this approach to checking if responses up to this point were sufficient?

Reply:
The idea is that we validate that we have sufficient information _from each shard_. So, if the keys for the shards covered by the node whose response we have received is considered sufficient for its ranges, then once we receive such responses from every shard we should have sufficient information altogether. We validate this before invoking the final callback by invoking `isSufficient(merged)`.

Reply:
Ok, so I think the thing I probably missed isSufficient is checked per shard. I have been living my life thinking we were checking for the whole thing.